### PR TITLE
CMQ-1347, CMQ-1348 and CMQ-1349 New email templates for fraud prevention header reports

### PR DESCRIPTION
**Pull request by edjannoo**

### CODE REVIEW

File: app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportHeuristicallyCompliant.scala.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        <p style="font-size: 19px; font-weight: bold; line-height: 1.315789474; margin: 0 0 30px 0;">@reportParams.applicationName</p>
+
+        <p style="font-size: 19px; line-height: 1.315789474; margin: 0 0 30px 0;">Dear @reportParams.developerName</p>
+
+        <p style="font-size: 19px; line-height: 1.315789474; margin: 0 0 30px 0;">
+            To help protect your customers' confidential data from criminals and fraudsters, we monitor API calls and send you feedback every month.
+        </p>
+
+        <p style="font-size: 19px; line-height: 1.315789474; margin: 0 0 30px 0;">
+            Our logs show that all of the fraud prevention headers submitted for @reportParams.applicationName between
```
Comment (czeczotka) [inline app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportHeuristicallyCompliant.scala.html:32]: Do we want to put quotes around application name?

File: app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportHeuristicallyCompliant.scala.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+@defining(TdqFphReportParams(params)) { reportParams =>
+VAT Making Tax Digital (MTD)
+
+@reportParams.applicationName
+
+Dear @reportParams.developerName
+
+To help protect your customers' confidential data from criminals and fraudsters, we monitor API calls and send you feedback every month.
+
+Our logs show that all of the fraud prevention headers submitted for @reportParams.applicationName between @reportParams.fromDate and @reportParams.toDate meet the current specification on the HMRC Developer Hub. Thank you, we'll send you a report if this changes.
```
Comment (czeczotka) [inline app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportHeuristicallyCompliant.scala.txt:12]: Do we want to put quotes around application name (here and below)?

File: app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportNonCompliant.scala.html [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                Use the Test API before you submit any headers. You need to check your data meets the specification and
+                is formatted correctly.
+            </li>
+        </ol>
+        <p style="font-size: 19px; line-height: 1.315789474; margin: 0 0 30px 0;">
+            If the Test API shows that you're submitting the correct headers, check if your end users are using an outdated
+            version of your application.
+        </p>
+        <p style="font-size: 19px; line-height: 1.315789474; margin: 0 0 30px 0;">
+            Most organisations are able to send all header data required for their connection method. If you’re unable to
```
Comment (czeczotka) [inline app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportNonCompliant.scala.html:77]: Wild apostrophe

File: app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportNonCompliant.scala.html [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+                    You need to correct errors @if(reportParams.hasWarnings){and consider warnings }
+                } else {
+                    You need to consider warnings
+                }
+            </h2>
+            <p style="font-size: 19px; line-height: 1.315789474; margin: 0 0 30px 0;">
+                We’ve made basic checks on header values sent by your application. This includes requests with all headers missing. It does not include headers that appear to be valid.
+            </p>
+            <p style="font-size: 19px; line-height: 1.315789474; margin: 0 0 30px 0;">
+                If your application sent requests via an additional connection method, you’ll receive a separate report.
```
Comment (czeczotka) [inline app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportNonCompliant.scala.html:93]: Wild apostrophe

File: app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportNonCompliant.scala.txt [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+Submit header data for every request made by your application.@if(reportParams.hasAllHeadersMissing) { Your application is missing all of the header data in @reportParams.allHeadersMissingPercentage% of requests.}
+
+@if(reportParams.hasErrors) {Correct the errors @if(reportParams.hasWarnings){and consider the warnings }shown in this report.}
+@if(reportParams.hasWarnings && !reportParams.hasErrors) {Consider the warnings shown in this report.}
+
+Use the Test API before you submit any headers. You need to check your data meets the specification and is formatted correctly.
+
+If the Test API shows that you're submitting the correct headers, check if your end users are using an outdated version of your application.
+
+Most organisations are able to send all header data required for their connection method. If you’re unable to submit a value, check what you need to do in the missing header section of the specification.
```
Comment (czeczotka) [inline app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportNonCompliant.scala.txt:31]: Wild apostrophe

File: app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportNonCompliant.scala.txt [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+@if(reportParams.hasErrors || reportParams.hasWarnings) {
+@if(reportParams.hasErrors) {You need to correct errors @if(reportParams.hasWarnings){and consider warnings}} else {You need to consider warnings}
+
+We’ve made basic checks on header values sent by your application. This includes requests with all headers missing. It does not include headers that appear to be valid.
```
Comment (czeczotka) [inline app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportNonCompliant.scala.txt:36]: Wild apostrophe

File: app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportNonCompliant.scala.txt [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+If your application sent requests via an additional connection method, you’ll receive a separate report.
```
Comment (czeczotka) [inline app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportNonCompliant.scala.txt:38]: Wild apostrophe

**Review by czeczotka — DISMISSED** [review]

File: app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportNonCompliant.scala.html (hunk 3 above)
Comment (edjannoo) [reply app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportNonCompliant.scala.html]: Copied and pasted from the Google Sheet :(
I've corrected this one and the others in this file and the `.txt` file too.

File: app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportHeuristicallyCompliant.scala.html (hunk 1 above)
Comment (edjannoo) [reply app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/tdqFphReportHeuristicallyCompliant.scala.html:32]: Mia's template and example don't use quotes around the application name

**Review by czeczotka — DISMISSED** [review]

**Review by czeczotka — APPROVED** [review]

**Comment by elabeca** [comment]
branch will need to be rebased with master before I approve. thank you.

**Review by elabeca — APPROVED** [review]

**Pull request merged**